RELEASE.md

```diff
@@ -1,23 +1,233 @@
-7. Publish the crate(s) on crates.io. For `scylla` crate, go directly to the crate directory, i.e. `/your/repo/path/scylla-rust-driver/scylla`, and run `cargo publish`. It will ask you for an access token, follow the instructions if you haven't set one up.
-8. Send the release notes to scylladb-users (https://groups.google.com/g/scylladb-users) as well as Cassandra users (https://lists.apache.org/list.html?user@cassandra.apache.org) mailing lists.
-9. Publish the documentation for the new version. In `docs/source/conf.py`,
-update the following:
-- `TAGS = [' ']` - add the new release tag to the list
-- `LATEST_VERSION = ' '` - replace the previous tag with the new release tag
+# Maintenance
+
+The following document includes information how to release scylla-rust-driver and
+other informations / procedures useful for maintainers.
+
+## Crates and versions
+
+This workspace consists of 4 crates:
+- `scylla`: main driver crate to be used by users. Depends on `scylla-cql` and `scylla-macros`.
+- `scylla-macros`: provides derive macros for users to be able to use their structs as UDTs / query parameters.
+- `scylla-cql`: low-level implementation of CQL binary protocol. Depends on `scylla-macros`.
+- `scylla-proxy`: mostly internal crate that aids testing. Depends on `scylla-cql`.
+
+We publish all 4 crates on [crates.io](https://crates.io/crates/scylla).
+
+Versions of the crates are decided with strict accordance to semver.
+For a sources on semver and semver in rust see:
+- https://doc.rust-lang.org/cargo/reference/resolver.html
+- https://doc.rust-lang.org/cargo/reference/semver.html
+- https://semver.org/
+- https://rust-lang.github.io/api-guidelines/necessities.html#public-dependencies-of-a-stable-crate-are-stable-c-stable (and some other parts of this book).
+
+What needs to be considered for releasing:
+- You can't change major version of scylla-cql / scylla-macros dependency in a minor update of scylla crate.
+For example, if scylla 0.13.0 depends on scylla-cql 0.4, then scylla 0.13.1 can't depend on scylla-cql 0.5.
+- We typically bump major versions of all 3 main crates together. I'm not sure if it's necessary - I remember there were
+arguments for that, but I can't remember them.
+
+
+## Documentation
+
+There are 2 places where documentation is published:
+- [docs.rs](https://docs.rs/scylla/latest/scylla/): API docs, generated automatically when publishing a crate to [crates.io.](https://crates.io/crates/scylla).
+- [Documentation book](https://rust-driver.docs.scylladb.com/stable/): generated and published by our CI (see [docs-pages.yaml](https://github.com/scylladb/scylla-rust-driver/blob/main/.github/workflows/docs-pages.yaml)) when there is a push to either `main` or some `branch-*` branch that updates something in `docs/` directory, or a push to some `v0.*` tag.
```

Comment:
`or some branch-* branch that updates something in `docs/` directory` - is this part true after the previous commit?